Automattic/jetpack
# VideoPress: Introduce private videos

Adds settings for controlling VideoPress video privacy. The settings work at two levels of granularity

### Site level

This is set from Jetpack Settings. When it's on, all video views are restricted to site members by default. Individual video privacy settings can be set at the video block level.

<img width="999" alt="Screenshot 2022-03-21 at 5 50 09 PM" src="https://user-images.githubusercontent.com/500744/159300001-3a90e61a-2d51-4419-a534-a0149c03f8d0.png">

### Video Level

An individual video's privacy setting can be changed from the block:

<img width="275" alt="Screenshot 2022-03-21 at 5 49 36 PM" src="https://user-images.githubusercontent.com/500744/159299915-1f3a10c4-fbb4-4b78-964d-9068eefccf5b.png">

The possible values are:

* Public (this specific video can be viewed by anyone).
* Private (this specific video can be viewed by site members).
* Site default (default for new videos, the video privacy setting follows the site-level setting).
<img width="283" alt="Screenshot 2022-03-21 at 5 49 55 PM" src="https://user-images.githubusercontent.com/500744/159299949-ced3d3e9-bcd7-4a0e-a27a-9cedcba96bcf.png">





#### Changes proposed in this Pull Request:

* Video Block: Adds Privacy setting Select field.
* Jetpack Settings: Adds a Site-wide default video privacy setting.

#### Jetpack product discussion

pxWta-1cj-p2


#### Does this pull request change what data or activity we track or use?
No



#### Testing instructions:

* On a sandboxed WPCOM, apply D75160-code
* It's better if this is tested on a test site with Jetpack VideoPress plan, since you are going to need to upload more than one videos.
* The default site-level privacy setting will be public.
* Create a new post, upload a video. Save and publish. Open post on a private tab. Video is visible.
* Edit post. On the video block, set Privacy setting to Private. Save. Open post on private tab. You get a message that the video is private.
* Go to Jetpack Settings and toggle site-wide private videos. Note: if you are testing on a wpcom simple site, You need to set the site privacy setting to private instead.
* Create a new post and upload a video. Save. Opening the post on a private window, you get a message that the video is private.
* Edit the post and set video privacy to public. Now the post video should play back on private windows.
* Finally, create a new post, and create 2 new video blocks. Set one to public and one to private. On Jetpack Settings, enable site-wide private videos. Viewing the post on a private browser window, one ofthe videos will be playing back, the other will display a private message.
<img width="538" alt="Screenshot 2022-03-21 at 5 54 12 PM" src="https://user-images.githubusercontent.com/500744/159300132-da66a7b4-0778-4a47-ac50-3b010dda8ac8.png">
